### PR TITLE
Sync yaml files

### PR DESCRIPTION
**Pull request by trask**
`pr.yaml`, `ci.yaml`, `nightly.yaml`, `nightly-no-cache.yaml`, `release-build.yml`, and `patch-release-build.yml` now all run:

* test
* testLatestDep (except for `pr.yaml` due to it causing contributors headaches sometimes when a new breaking library version is released)
* smoke-tests
* example-distro

### CODE REVIEW

File: .github/workflows/pr.yaml [hunk 1]
```diff
@@ -77,6 +78,10 @@ jobs:
           path: /tmp/deadlock-detector-*
           if-no-files-found: ignore
 
+  # testLatestDep is intentionally not included in the PR workflow
+  # because it can break randomly when a new library version is released
+  # which can be confusing for contributors
```
Comment (trask) [inline .github/workflows/pr.yaml:83]: I'm wondering how this is different from running muzzle in the PR workflow..

File: .github/workflows/nightly-no-cache.yaml [hunk 2]
```diff
@@ -50,6 +50,22 @@ jobs:
       - name: Test
         run: ./gradlew test -PtestJavaVersion=${{ matrix.java }} --stacktrace -x :smoke-tests:test -Porg.gradle.java.installations.paths=${{ steps.setup-test-java.outputs.path }} -Porg.gradle.java.installations.auto-download=false --no-build-cache
 
+  testLatestDep:
```
Comment (iNikem) [inline .github/workflows/nightly-no-cache.yaml:53]: I did not include this job into this workflow on purpose. Assuming that latestDep test does not use cache anyway.

File: .github/workflows/patch-release-build.yml [hunk 3]
```diff
@@ -23,39 +23,155 @@ jobs:
...
+
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        java: [ 8, 11, 15 ]
+    steps:
+      - uses: actions/checkout@v2.3.4
+        with:
+          ref: ${{ needs.prepare-release-branch.outputs.release-branch-name }}
```
Comment (iNikem) [inline .github/workflows/patch-release-build.yml:57]: If you reference `needs` here, shouldn't this job actually have a `needs` section?

File: .github/workflows/pr.yaml (hunk 1 above)
Comment (mateuszrzeszutek) [reply .github/workflows/pr.yaml:83]: Muzzle only runs for the instrumentation code; latestDepTest often detects problems in the test code (or slight changes in emitted telemetry)

Comment (trask) [reply .github/workflows/pr.yaml]: 👍 updated comment to reflect

File: .github/workflows/nightly-no-cache.yaml (hunk 2 above)
Comment (trask) [reply .github/workflows/nightly-no-cache.yaml:53]: > Assuming that latestDep test does not use cache anyway.

I'll test that out 👍

File: .github/workflows/gradle-wrapper-validation.yml [hunk 4]
```diff
@@ -13,7 +13,7 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2.3.4
+
+      - uses: gradle/wrapper-validation-action@v1.0.4
         with:
           fetch-depth: 0
```
Comment (mateuszrzeszutek) [inline .github/workflows/gradle-wrapper-validation.yml:19]: Isn't `fetch-depth` a checkout (git clone) option?

Comment (trask) [reply .github/workflows/gradle-wrapper-validation.yml:17]: 😱 thanks!

**Review by mateuszrzeszutek — APPROVED** [review]

File: .github/workflows/nightly-no-cache.yaml (hunk 2 above)
Comment (trask) [reply .github/workflows/nightly-no-cache.yaml:53]: It looks like `testLatestDeps` does use the cache?

```
$ ./gradlew :instrumentation:spring:spring-webflux-5.0:javaagent:test -PtestLatestDeps=true

...

BUILD SUCCESSFUL in 1m 19s
68 actionable tasks: 5 executed, 63 up-to-date
S3 cache 300ms wasted on misses, reads: 3, elapsed: 300ms

$ ./gradlew :instrumentation:spring:spring-webflux-5.0:javaagent:test -PtestLatestDeps=true

...

BUILD SUCCESSFUL in 6s
68 actionable tasks: 68 up-to-date
```

Comment (iNikem) [reply .github/workflows/nightly-no-cache.yaml:53]: Interesting... I wonder if Gradle invalidates cache if new version is detected... But anyway it does make sense to include it in this workflow 👍

**Review by iNikem — APPROVED** [review]

**Pull request merged**